src/promptflow-tools/promptflow/tools/aoai_gpt4v.py

```diff
@@ -7,12 +7,130 @@
 from promptflow._internal import ToolProvider, tool
 from promptflow.connections import AzureOpenAIConnection
 from promptflow.contracts.types import PromptTemplate
+from typing import List, Dict
 
 from promptflow.tools.common import render_jinja_template, handle_openai_error, parse_chat, \
     preprocess_template_string, find_referenced_image_set, convert_to_chat_list, normalize_connection_config, \
     post_process_chat_api_response
 
 
+GPT4V_VERSION = "vision-preview"
+
+
+def _get_credential():
+    from azure.identity import DefaultAzureCredential
+    from azure.ai.ml._azure_environments import _get_default_cloud_name, EndpointURLS, _get_cloud, AzureEnvironments
```

Comment:
private methods are ususally unstable, could we change to use more trustful method?

Reply:
No idea. It's same with SDK's solution currently.